core/request.go

```diff
@@ -130,6 +130,23 @@ func (r *RequestAccessor) ProxyEventToHTTPRequest(req events.APIGatewayProxyRequ
 			queryCnt++
 		}
 	}
+	if len(req.MultiValueQueryStringParameters) > 0 {
+		queryDelim := "&"
+		queryStart := false
+		if queryString == "" {
+			queryDelim = "?"
+			queryStart = true
+		}
+		for q, l := range req.MultiValueQueryStringParameters {
+			for _, v := range l {
+				queryString += queryDelim + url.QueryEscape(q) + "=" + url.QueryEscape(v)
+				if queryStart {
```

Comment:
Why a separate variable, you could just check if the `queryDelim` == `?`. Or even better, just create the string always using the `&` as separator and then append the `?` when concatenating at the end. url + "?" + queryString

Reply:
You're saying `?` should be added regardless if there are query params or not? I.e. you think something like `/path?` should be the result? And if query params are provided, something like `/path?param1=abc&param2=def&` (or `/path?&param1=abc&param2=def`) should be the result? They are of course valid from a URI spec perspective, if I recall correctly. Or am I misunderstanding you?

Reply:
To keep it simple, I would do this (not checked in code editor so there may be mistakes):

```go
path := req.Path + 
if len(req.MultiValueQueryStringParameters) > 0 {
  queryString := ""
  for q, l := range req.MultiValueQueryStringParameters {
    for _, v := range l {
      if queryString != "" {
        queryString += "&"
      }
      queryString += url.QueryEscape(q) + "=" + url.QueryEscape(v)
    }
  }
  path += "?" + queryString
}
```